MeesGame/Gameplay/GameObjects/TilesTypes/PortalTile.cs

```diff
@@ -69,11 +69,29 @@ public override void UpdateToAdditionalInfo()
         /// </summary>
         public override void UpdatePortals()
         {
-            foreach (Tile tile in this.TileField.Objects)
+            int tileFieldWitdth = TileField.Objects.GetLength(0);
+            int tileFieldHeight = TileField.Objects.GetLength(1);
+            int maxTileIndex = tileFieldHeight * tileFieldWitdth;
+
+            int myIndex = Location.X + tileFieldWitdth * Location.Y;
+
+            int targetIndex = myIndex +1;
+
+            while (true)
             {
-                if (tile.Data.AdditionalInfo == portalIndex && tile != this && tile.TileType == TileType.Portal)
+                if(maxTileIndex > targetIndex)
+                {
+                    Tile targetTile = TileField.Objects[targetIndex % tileFieldWitdth, targetIndex / tileFieldWitdth] as Tile;
+                    if (targetTile.Data.AdditionalInfo == portalIndex && targetTile != this && targetTile.TileType == TileType.Portal)
+                    {
+                        destination = targetTile.Location;
+                        return;
+                    }
+                    targetIndex++;
+                }
+                else
                 {
-                    destination = tile.Location;
+                    targetIndex = 0;
```

Comment:
Right now the game crashes if you try to load a level with a portal that has no destination.